stan/math/opencl/sub_block.hpp

```diff
@@ -57,7 +57,35 @@ inline void matrix_cl::sub_block(const matrix_cl& A, size_t A_i, size_t A_j,
   } else {
     opencl_kernels::sub_block(cl::NDRange(nrows, ncols), A, *this, A_i, A_j,
                               this_i, this_j, nrows, ncols, A.rows(), A.cols(),
-                              this->rows(), this->cols(), triangular_view);
+                              this->rows(), this->cols(), A.triangular_view());
+  }
+  // calculation of extreme sub- and super- diagonal written
+  int diag_in_copy = A_i - A_j;
+  int copy_low = static_cast<bool>(A.triangular_view_ & TriangularViewCL::Lower)
```

Comment:
> Why not use A.triangular_view() here?

This is a method of matrix_cl so it can access private attribute. If for some reason going through getter is preferable I can change that.

> Why is this being cast to a bool and then implicitly an int?

It is not. bool is used in ternary operator on the next line.